xwiki-commons-core/xwiki-commons-configuration/xwiki-commons-configuration-default/src/main/java/org/xwiki/configuration/internal/SystemPropertiesConfigurationSource.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.configuration.internal;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.configuration.ConfigurationSaveException;
+
+/**
+ * {@link System} properties based configuration source.
+ *
+ * @version $Id$
+ * @since 17.4.0RC1
+ */
+@Component
+@Singleton
+@Named(SystemPropertiesConfigurationSource.HINT)
+public class SystemPropertiesConfigurationSource extends AbstractPropertiesConfigurationSource
+{
+    /**
+     * The hint to use to get this configuration source.
+     */
+    public static final String HINT = "systemproperties";
+
+    /**
+     * The prefix used to identify system properties used to overwrite the configuration.
+     */
+    public static final String PREFIX = "xconf.";
+
+    String getSystemProperty(String key)
+    {
+        return System.getProperty(toSystemKey(key));
+    }
+
+    Properties getSystemProperties()
+    {
+        return System.getProperties();
+    }
+
+    String setSystemProperty(String key, String value)
+    {
+        return System.setProperty(key, value);
```

Comment:
no check on the key for the prefix?

Reply:
I'd have expected that you reuse `fromSystemKey` and that it would check the presence of the prefix.

Reply:
As mentioned above, those methods are just here to be overwritten in tests.